### PR TITLE
[bug](be) fix accept null predicate mem leak

### PR DESCRIPTION
**Pull request by ByteYue**
# Proposed changes

Issue Number: close #xxx
The former logic inside accept null predicate would cause memory leak.
## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by ByteYue** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

**Review by xiaokang — DISMISSED** [review]
LGTM

File: be/src/olap/accept_null_predicate.h [hunk 1]
```diff
@@ -60,7 +60,8 @@ class AcceptNullPredicate : public ColumnPredicate {
                       bool* flags) const override {
         if (column.has_null()) {
             // copy original flags
-            bool* original_flags = new bool[size];
+            auto original_flags_buf = std::make_unique<bool>(size);
```
Comment (gavinchou) [inline be/src/olap/accept_null_predicate.h:63]: Should it be `std::make_unique<bool[]>(size)`?

File: be/src/olap/accept_null_predicate.h [hunk 2]
```diff
@@ -125,7 +126,8 @@ class AcceptNullPredicate : public ColumnPredicate {
                           bool* flags) const override {
         if (column.has_null()) {
             // copy original flags
-            bool* original_flags = new bool[size];
+            auto original_flags_buf = std::make_unique<bool>(size);
```
Comment (gavinchou) [inline be/src/olap/accept_null_predicate.h:129]: Ditto

**Review by gavinchou — CHANGES_REQUESTED** [review]

**Comment by ByteYue** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**